src/Libraries/Microsoft.Extensions.Telemetry.Abstractions/Metrics/CommonTagAttribute.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.Shared.Diagnostics;
+
+namespace Microsoft.Extensions.Telemetry.Metrics;
+
+/// <summary>
+/// Provides a way to add common tags to the all the metrics in this class.
+/// </summary>
+/// <example>
+/// <code language="csharp">
+/// [CommonTag("Category", "Http")]
+/// [CommonTag("MetricVersion", "v1")]
+/// static partial class Metric
+/// {
+///     [Counter("RequestName", "RequestStatusCode")]
+///     static partial RequestCounter CreateRequestCounter(Meter meter);
+/// }
+/// </code>
+/// </example>
+[AttributeUsage(AttributeTargets.Class, AllowMultiple = true)]
+[Conditional("CODE_GENERATION_ATTRIBUTES")]
+[Experimental("EXPERIMENTAL_COMMON_TAG_ATTRIBUTE")]
```

Comment:
Should use the well-defined experiment IDs

```suggestion
[Experimental(diagnosticId: DiagnosticIds.Experiments.Telemetry, UrlFormat = DiagnosticIds.UrlFormat)]
```